sql/mysqld.cc

```diff
@@ -6316,6 +6529,9 @@ void handle_connections_sockets()
     create_new_thread(thd);
     set_current_thd(0);
   }
+#ifdef HAVE_SYSTEMD
+  sd_notify(0, "STOPPING=1");
```

Comment:
Okay. Indeed, now that I'm thinking about it, in almost all cases of "MySQL/MariaDB is unavailable" that I've seen, the process didn't really hang. So a simple built-in watchdog cannot detect when dbms "doesn't respond" from the user point of view.